weave/flow/chat_util.py

```diff
@@ -22,75 +22,83 @@ def __init__(self, chunk_iter: Iterator[ChatCompletionChunk]) -> None:
         self.first_chunk: Optional[ChatCompletionChunk] = None
         self.output_choices: list[dict] = []
 
+    def __next__(self) -> ChatCompletionChunk:
+        chunk = self.chunk_iter.__next__()
+        self._process_chunk(chunk)
+        return chunk
+
     def __iter__(self) -> Iterator[ChatCompletionChunk]:
         for chunk in self.chunk_iter:
+            self._process_chunk(chunk)
             yield chunk
-            if self.first_chunk is None:
-                self.first_chunk = chunk
-            for chunk_choice in chunk.choices:
-                for i in range(chunk_choice.index + 1 - len(self.output_choices)):
-                    self.output_choices.append(
-                        {
-                            "index": len(self.output_choices),
-                            "message": {
-                                "content": None,
-                                "tool_calls": None,
-                            },
-                            "finish_reason": None,
-                            "logprobs": None,
-                        }
-                    )
 
-                # choice fields
-                choice = self.output_choices[chunk_choice.index]
-                if chunk_choice.finish_reason:
-                    choice["finish_reason"] = chunk_choice.finish_reason
-                if chunk_choice.logprobs:
-                    choice["logprobs"] = chunk_choice.logprobs
+    def _process_chunk(self, chunk: ChatCompletionChunk) -> None:
```

Comment:
This looks like a huge change, but if you hide whitespace, you see that it is just moving some logic to a helper function.